specifyweb/frontend/js_src/lib/components/QueryBuilder/fieldSpec.ts

```diff
@@ -37,6 +37,14 @@ const reDatePart = /(.*)(NumericDay|NumericMonth|NumericYear)$/;
 
 export type DatePart = 'day' | 'fullDate' | 'month' | 'year';
 
+const lowercaseTrees = [
```

Comment:
Even better, you can use this helper: https://github.com/specify/specify7/blob/26c02a287ad6c9aa2d83834e20f992fe51830c8a/specifyweb/frontend/js_src/lib/components/InitialContext/treeRanks.ts#L61

Reply:
> Can you instead export this variable and use it here?
> 
> https://github.com/specify/specify7/blob/26c02a287ad6c9aa2d83834e20f992fe51830c8a/specifyweb/frontend/js_src/lib/components/InitialContext/treeRanks.ts#L56

Actually, since it pulls the field name from the base table, there are more fields that would need to be checked for:
- Paleocontext
  - bioStrat
  - chronosStrat
  - chronosStratEnd
  
- CollectingEventAttribute
  - hostTaxon 
  
- Determination
  - preferredTaxon
  
- Taxon
  - acceptedChildren
  - acceptedTaxon
  - children
  - hybridChildren1
  - hybridChildren2
  - hybridParent1
  - hybridParent2
  - parent  
  
- Taxon Attributes
  - taxons 
  
- Tree Definition (& Tree Definition Item)
  - treeEntries 
   
- Geography
  - acceptedChildren
  - acceptedGeography
  - parent  
  
- Preparation
  - alternateStorage 
  
- Storage
  - acceptedChildren
  - acceptedStorage
  - children
  - parent

And these are just fields for Paleocontext, and all relationship field names for the Geography, Taxon, and Storage trees.  Perhaps a better implementation would utilize getting the SpecifyModel name...

Reply:
> Even better, you can use this helper:
> 
> https://github.com/specify/specify7/blob/26c02a287ad6c9aa2d83834e20f992fe51830c8a/specifyweb/frontend/js_src/lib/components/InitialContext/treeRanks.ts#L61

I was considering `isTreeModel`, however `isTreeModel` is case sensitive and many of the tree fields do not match the correct case (and unless we have some helper function, coercing the strings to the correct case may be difficult)

Reply:
Fixed in 129d825

Reply:
Although this still may fail for Chronostratigraphy trees. Should this be added in the `treesForPaleo` variable?